array_transient default constructor produces compilation error

I have a file with contents:

```cpp
#include <immer/array_transient.hpp>

void f() {
    immer::array_transient<int> x;
}
```

This should be a supported use case:

https://github.com/arximboldi/immer/blob/master/immer/array_transient.hpp#L51-L56

The MSVC2017 compiler produces the following confusing error, which does not point to the line causing the error:

```
C:\Users\nyanpasu\Dropbox\encrypted\code\exotracker\3rdparty\immer/array_transient.hpp(189): error C2664: 'immer::detail::arrays::with_capacity<T,MemoryPolicy>::with_capacity(immer::detail::arrays::with_capacity<T,MemoryPolicy> &&)': cannot convert argument 1 from 'const immer::detail::arrays::with_capacity<T,MemoryPolicy> &(__cdecl *)(void)' to 'const immer::detail::arrays::no_capacity<T,MemoryPolicy> &'
        with
        [
            T=int,
            MemoryPolicy=immer::default_memory_policy
        ]
C:\Users\nyanpasu\Dropbox\encrypted\code\exotracker\3rdparty\immer/array_transient.hpp(189): note: Reason: cannot convert from 'overloaded-function' to 'const immer::detail::arrays::no_capacity<T,MemoryPolicy>'
        with
        [
            T=int,
            MemoryPolicy=immer::default_memory_policy
        ]
C:\Users\nyanpasu\Dropbox\encrypted\code\exotracker\3rdparty\immer/array_transient.hpp(189): note: No constructor could take the source type, or constructor overload resolution was ambiguous
C:\Users\nyanpasu\Dropbox\encrypted\code\exotracker\3rdparty\immer/detail/arrays/with_capacity.hpp(50): note: see declaration of 'immer::detail::arrays::with_capacity<T,MemoryPolicy>::with_capacity'
        with
        [
            T=int,
            MemoryPolicy=immer::default_memory_policy
        ]
```

I'm building with immer master 0c174600d9532254d184098d6c521d8c17f44ae1 plus a few changes to fix https://github.com/arximboldi/immer/issues/11#issuecomment-555072407 and silence warnings. I have not tested other compilers.

On MinGW 7.3.0 with g++ 7.3.0, I get the following error which correctly pinpoints the line in my source file causing the issue:

```
C:\Users\nyanpasu\Dropbox\encrypted\code\exotracker\src\audio\synth\sequencer.cpp: In function 'void f()':
C:\Users\nyanpasu\Dropbox\encrypted\code\exotracker\src\audio\synth\sequencer.cpp:4:33: error: could not convert 'immer::detail::arrays::with_capacity<int, immer::memory_policy<immer::free_list_heap_policy<immer::cpp_heap>, immer::refcount_policy> >::empty' from 'const immer::detail::arrays::with_capacity<int, immer::memory_policy<immer::free_list_heap_policy<immer::cpp_heap>, immer::refcount_policy> >&()' to 'immer::array_transient<int>::impl_t {aka immer::detail::arrays::with_capacity<int, immer::memory_policy<immer::free_list_heap_policy<immer::cpp_heap>, immer::refcount_policy> >}'
     immer::array_transient<int> x;
                                 ^
In file included from C:\Users\nyanpasu\Dropbox\encrypted\code\exotracker\src\audio\synth\sequencer.cpp:1:0:
C:/Users/nyanpasu/Dropbox/encrypted/code/exotracker/3rdparty/immer/array_transient.hpp: In constructor 'constexpr immer::array_transient<T, MemoryPolicy>::array_transient() [with T = int; MemoryPolicy = immer::memory_policy<immer::free_list_heap_policy<immer::cpp_heap>, immer::refcount_policy>]':
C:/Users/nyanpasu/Dropbox/encrypted/code/exotracker/3rdparty/immer/array_transient.hpp:56:5: error: could not convert 'immer::detail::arrays::with_capacity<int, immer::memory_policy<immer::free_list_heap_policy<immer::cpp_heap>, immer::refcount_policy> >::empty' from 'const immer::detail::arrays::with_capacity<int, immer::memory_policy<immer::free_list_heap_policy<immer::cpp_heap>, immer::refcount_policy> >&()' to 'immer::array_transient<int>::impl_t {aka immer::detail::arrays::with_capacity<int, immer::memory_policy<immer::free_list_heap_policy<immer::cpp_heap>, immer::refcount_policy> >}'
     array_transient() = default;
     ^~~~~~~~~~~~~~~
C:\Users\nyanpasu\Dropbox\encrypted\code\exotracker\src\audio\synth\sequencer.cpp: In function 'void f()':
C:\Users\nyanpasu\Dropbox\encrypted\code\exotracker\src\audio\synth\sequencer.cpp:4:33: note: synthesized method 'constexpr immer::array_transient<T, MemoryPolicy>::array_transient() [with T = int; MemoryPolicy = immer::memory_policy<immer::free_list_heap_policy<immer::cpp_heap>, immer::refcount_policy>]' first required here 
     immer::array_transient<int> x;
                                 ^
```

--- a/immer/array_transient.hpp
+++ b/immer/array_transient.hpp
@@ -186,7 +186,7 @@ private:
         : impl_(std::move(impl))
     {}
 
-    impl_t  impl_  = impl_t::empty;
+    impl_t  impl_  = impl_t::empty();
 };
 
 } // namespace immer

--- a/test/array_transient/default.cpp
+++ b/test/array_transient/default.cpp
@@ -14,6 +14,11 @@
 
 #include "../vector_transient/generic.ipp"
 
+TEST_CASE("array_transient default constructor compiles")
+{
+    immer::array_transient<int> transient;
+}
+
 TEST_CASE("array provides mutable data")
 {
     auto arr = immer::array<int>(10, 0);

